tests/ixia/test_ixia_traffic_restpy.py

```diff
@@ -0,0 +1,153 @@
+from common.reboot import *
+from ixnetwork_restpy import SessionAssistant, Files
+
+from common.ixia.ixia_fixtures import ixia_api_serv_ip, ixia_api_serv_user,\
+     ixia_api_serv_passwd, ixia_api_serv_port, ixia_api_serv_session_id, \
+     ixia_api_server_session  
+
+from common.ixia.ixia_helpers import get_neigh_ixia_mgmt_ip, get_neigh_ixia_card,\
+     get_neigh_ixia_port, IxiaFanoutManager
+
+import time
+
+def create_ipv4_traffic_end_points (
+    src_start_port,
+    src_port_count,
+    src_first_route_index,
+    src_route_count,
+    dst_start_port,
+    dst_port_count,
+    dst_first_route_index,
+    dst_route_count) :
+
+    src = [{'arg1': '/api/v1/sessions/1/ixnetwork/topology/1/deviceGroup/1/ethernet/1/ipv4/1',
+             'arg2': src_start_port, 
+             'arg3': src_port_count, 
+             'arg4': src_first_route_index, 
+             'arg5': dst_route_count}
+    ]
+
+    dst = [{'arg1': '/api/v1/sessions/1/ixnetwork/topology/1/deviceGroup/1/ethernet/1/ipv4/1', 
+             'arg2': dst_start_port, 
+             'arg3': dst_port_count, 
+             'arg4': dst_first_route_index,
+             'arg5': dst_route_count}
+    ]
+
+    return (src, dst)
+
+
+def test_testbed(testbed, conn_graph_facts, duthost, fanout_graph_facts,
+    ixia_api_server_session, fanouthosts):
+
+    logger.info("Connection Graph Facts = %s " %(conn_graph_facts))
+    logger.info("Fanout Graph facts = %s" %(fanout_graph_facts))
+    logger.info("DUT hostname = %s" %(duthost.hostname))
+ 
+    mg_facts  = duthost.minigraph_facts(host=duthost.hostname)
+    gatewayIp = mg_facts['ansible_facts']['minigraph_vlan_interfaces'][0]['addr']
+
+    ixiaFanoutHostList = IxiaFanoutManager(fanout_graph_facts) 
+    ixiaFanoutHostList.get_fanout_device_details(device_number = 0)
+
+    # Build gateway valuelist. Same gateway IP for all interface
+    gateway_value_list = []
+    for i in ixiaFanoutHostList.ports() :
+        gateway_value_list.append(gatewayIp)
+
+    # Create ixNetwork interface IP address list 
+    interface_ip_list = []
+    ipaddress = ipaddr.IPv4Address(gatewayIp)
+    for i in ixiaFanoutHostList.ports() :
+        ipaddress = ipaddress + 1
+        interface_ip_list.append(ipaddress._string_from_ip_int(ipaddress._ip))
+
+    session   = ixia_api_server_session
+    ixNetwork = session.Ixnetwork
+    portMap   = session.PortMapAssistant()
+
+    vport_list = []
+    for i in ixiaFanoutHostList.ports() :
+        (chassisIp, cardId, portId) = ixiaFanoutHostList.getCardPort(i)
+        vport_list.append(portMap.Map(chassisIp, cardId, portId))
+       
+    t1 = time.time()
+    portMap.Connect(ChassisTimeout=1200, ForceOwnership=True)
+    t2 = time.time()
+
+    time_taken = t2 - t1
+    logger.info("time-taken to connect = %s" %(time_taken))
+
+    for vport in vport_list :
+        vport.L1Config.NovusHundredGigLan.IeeeL1Defaults        = False         
```

Comment:
There is still a lot of low level code in here. Please abstract them. User should just need to specify the num of ports needed for test, any traffic type that he wants, start and stop traffic. The rest of the information should all go into ixia libraries

Reply:
All the low-level functions are moved in the low level library